email/finalizer.go

```diff
@@ -230,3 +224,62 @@ func (f *Finalizer) threadedFinalizeMessages() {
 		}
 	}
 }
+
+// sendAbuseReport sends the abuse report for the given abuse email to the given
+// email address. This is extracted in a standalone function for unit testing
+// purposes.
+func sendAbuseReport(client *client.Client, email database.AbuseEmail, mailbox, to string) error {
+	// generate a uuid as message id
+	var u *uuid.UUID
+	u, err := uuid.NewV4()
+	if err != nil {
+		return errors.AddContext(err, "failed to generate uid")
+	}
+
+	// construct the email message
+	msg := fmt.Sprintf("Subject: Re: %s\n", email.Subject)
+	msg += fmt.Sprintf("Message-ID: <%s@abusescanner>\n", u)
+	msg += fmt.Sprintf("References: %s\n", email.MessageID)
+	msg += fmt.Sprintf("In-Reply-To: %s\n", email.MessageID)
+	msg += fmt.Sprintf("From: SCANNED <%s>\n", scannerEmailAddress)
+	msg += fmt.Sprintf("To: %s\n", to)
+	msg += ""
+	msg += email.String()
+	reader := strings.NewReader(msg)
+
+	// append an email with the abuse report result
+	err = client.Append(mailbox, nil, time.Now().UTC(), reader)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// sendAutomatedReply sends the automated reply for the given abuse email to the
+// original email sender. This is extracted in a standalone function for unit
+// testing purposes.
+func sendAutomatedReply(auth smtp.Auth, email database.AbuseEmail) error {
+	// generate a uuid as message id
+	var u *uuid.UUID
+	u, err := uuid.NewV4()
+	if err != nil {
+		return errors.AddContext(err, "failed to generate uid")
+	}
+
+	// construct the email message
+	msg := fmt.Sprintf("Subject: Re: %s\n", email.Subject)
+	msg += fmt.Sprintf("Message-ID: <%s@abusescanner>\n", u)
+	msg += fmt.Sprintf("References: %s\n", email.MessageID)
+	msg += fmt.Sprintf("In-Reply-To: %s\n", email.MessageID)
+	msg += fmt.Sprintf("From: <%s>\n", email.To)
+	msg += fmt.Sprintf("To:%s\n", email.Sender())
+	msg += ""
+	msg += email.Response()
+
+	// send the automated response
+	err = smtp.SendMail("smtp.gmail.com:587", auth, email.To, []string{email.Sender()}, []byte(msg))
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
```suggestion
	return smtp.SendMail("smtp.gmail.com:587", auth, email.To, []string{email.Sender()}, []byte(msg))
```

Reply:
I'll change it to what you suggest but I don't do this on purpose. I kind of like the error-equals-if-not -nil-return-else-return-nil-check. Mainly because if you want to sprinkle debug lines you don't have to make changes aside from your println additions.

Reply:
It's four lines of code that do exactly nothing. It's not a big deal, it's just bugging me.